app/controllers/tickets_controller.rb

```diff
@@ -1,6 +1,43 @@
 # frozen_string_literal: true
 
 class TicketsController < ApplicationController
+  include Rails::Pagination
+
+  SORT_WEIGHT_LAMBDAS = {
+    createdAt:
+      lambda { |ticket| ticket.created_at },
+  }.freeze
+
+  def index
+    tickets = Ticket
+
+    # Filter based on params
+    type = params[:type]
+    if type
+      tickets = tickets.where(metadata_type: type)
+    end
+
+    status = params[:status]
+    tickets = tickets.select do |ticket|
+      if status
+        ticket.metadata.status == status
+      else
+        true
+      end
+    end
+
+    # Filter based on current_user's permission
+    tickets = tickets.select do |ticket|
+      ticket.can_user_access?(current_user)
+    end
+
+    # Sort
+    sort_param = params[:sort] || ''
+    tickets = sort(tickets, sort_param, SORT_WEIGHT_LAMBDAS)
```

Comment:
Where is this `sort` method coming from?

Reply:
It's the same `sort_helper` we used for roles.